jerry-core/jrt/jrt.h

```diff
@@ -73,114 +80,57 @@
 #define JERRY_STATIC_ASSERT(x, msg) \
   enum { JERRY_STATIC_ASSERT_GLUE (static_assertion_failed_, __LINE__, msg) = 1 / (!!(x)) }
 
-/**
- * Variable that must not be referenced.
- *
- * May be used for static assertion checks.
- */
-extern uint32_t jerry_unreferenced_expression;
-
-extern void __noreturn jerry_assert_fail (const char *, const char *, const char *, const uint32_t);
-extern void __noreturn jerry_unreachable (const char *, const char *, const char *, const uint32_t);
-extern void __noreturn jerry_unimplemented (const char *, const char *, const char *, const uint32_t);
-
 #ifndef JERRY_NDEBUG
-#define JERRY_ASSERT(x) do { if (__builtin_expect (!(x), 0)) { \
-    jerry_assert_fail (#x, __FILE__, __func__, __LINE__); } } while (0)
-#else /* JERRY_NDEBUG */
-#define JERRY_ASSERT(x) do { if (false) { (void)(x); } } while (0)
-#endif /* !JERRY_NDEBUG */
-
-#define JERRY_UNUSED(x) ((void) (x))
+extern void __noreturn jerry_assert_fail (const char *, const char *, const char *, const uint32_t);
+extern void __noreturn jerry_unreachable (const char *, const char *, const uint32_t);
 
-#ifdef JERRY_ENABLE_LOG
-#define JERRY_DLOG(...) jerry_port_log (JERRY_LOG_LEVEL_DEBUG, __VA_ARGS__)
-#define JERRY_DDLOG(...) jerry_port_log (JERRY_LOG_LEVEL_TRACE, __VA_ARGS__)
-#else /* !JERRY_ENABLE_LOG */
-#define JERRY_DLOG(...) \
+#define JERRY_ASSERT(x) \
   do \
   { \
-    if (false) \
+    if (unlikely (!(x))) \
```

Comment:
Don't let be misguided by the misalignment of the diff. Unfortunately, the diff incorrectly compares `JERRY_ASSERT` to `JERRY_LOG`, must be becose of the movements of the code. moreover, this is debug code (`#ifndef JERRY_NDEBUG`) so no effect on release, and `JERRY_ASSERT` was always using `__builtin_expect(..., 0)` in its `if`. (See the rewritten `JERRY_DLOG` at line 123.)